js/rebase-client/src/lib.rs

```diff
@@ -33,40 +34,56 @@ pub struct Client {
     client: Arc<RebaseClient>,
 }
 
+#[wasm_bindgen]
+#[derive(Deserialize, Serialize)]
+pub struct Config {
+    instructions: String,
+    statement: String,
+    jwt: Option<String>,
+    ld: Option<String>,
+    verify_jwt: Option<String>,
+    verify_ld: Option<String>,
+}
+
+impl Config {
+    fn is_valid(&self) -> Result<(), String> {
+        if self.jwt.is_none() && self.ld.is_none() {
+            Err("At least one of JWT or LD urls must be set".to_string())
+        } else {
+            Ok(())
+        }
+    }
```

Comment:
Personally I think using an enum to encode this constraint would be preferable, perhaps I lean too far into "make incorrect things impossible":
```suggestion
        #[serde(flatten)]
        verifyOpts: VerifyOpts
    }

pub enum VerifyOpts {
    Jwt {
        jwt: Url,
        verify_jwt: Option<Url>
    },
    Ld {
        ld: Url,
        verify_ld: Option<Url>
    },
    Both {
        jwt: Url,
        ld: Url,
        verify_jwt: Option<Url>,
        verify_ld: Option<Url>
    }
}
```

Also, with the `serde` feature on the `url` crate, you can use the `Url` type here and it should deserialise